mat64/dense_arithmetic.go

```diff
@@ -351,6 +351,7 @@ func (m *Dense) Dot(b Matrix) float64 {
 }
 
 // Mul takes the matrix product of a and b, placing the result in the receiver.
+// If a or b are also the receiver, a workspace will be allocated.
```

Comment:
We can just say "temporary memory will be allocated". The word "workspace" is confusing to me in this context. 

If/when you revive the pools, could you add more comments? I'm better at Go than when it was first proposed, but I remember not understanding what was going on.